strax/context.py

```diff
@@ -977,27 +944,11 @@ def get_iter(self, run_id: str,
                         keep_columns=keep_columns,
                         time_range=time_range,
                         time_selection=time_selection)
-                    
-                    if progress_bar:
-                        # Update progressbar:
-                        if end_time - start_time > 0:
-                            pbar.n = (result.end - start_time) / (end_time - start_time)
-                        else:
-                            # Strange, start and endtime are the same, probably we 
-                            # don't have data yet e.g. allow_incomplete == True.
-                            pbar.n = 0
-                        pbar.update(0)
-                        # Now get last time printed and refresh seconds_per_chunk:
-                        # This is a small work around since we do not know the
-                        # pacemaker here and therefore we do not know the number of
-                        # chunks.
-                        sec_per_chunk = pbar.last_print_t - last_time
-                        pbar.postfix[1]['spc'] = sec_per_chunk
-                        pbar.postfix[1]['n'] = n_chunks
-                        pbar.refresh()
-                        last_time = pbar.last_print_t
+                    self._update_progress_bar(
+                        pbar, t_start, t_end, n_chunks, result.end)
 
                     yield result
+            _p.close()
```

Comment:
Needed? Should be close by the `with`-statement is not it?

Reply:
It's the tqdm recommended way according to their documentation. Felt like it did not hurt to do so:
https://github.com/tqdm/tqdm#manual